### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by carreraprogrammer**
This  branch was created in order to satisfy the next project's requirements

## Project Requirements

-  ### Create a class Classroom with the following

- [x] @label instance variable, should be initialized in the constructor.

- [x]  Setter and getter for @label (remember about attr_accessor).

- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented

- [x] Create the has-many side (a classroom has many students).

- [x] Create the belongs-to side (a student belongs to a classroom).

- [x] Make sure that when adding a student to a classroom it also sets the classroom for the student.

- [x] Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- Create a class Book with the following

- [x] @title and @author instance variables, should be initialized in the constructor.

- [x] Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following

- [x] @date instance variable, should be initialized in the constructor.

- [x] Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

- [x]  Create the has-many side of Book and Rental (a book has many rentals).

- [x]  Create the belongs-to side of Rental and Book (a rental belongs to a book).

- [x]  Create the has-many side of Person and Rental (a person has many rentals).

- [x]  Create the belongs-to side of Rental and Person (a rental belongs to a person).

- [x]  Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 require './nameable'
 require './capitalize_decorator'
 require './trimmer_decorator'
+require './rental'
 
 class Person < Nameable
   attr_reader :id
```
Comment (Ismailco) [inline person.rb:7]: - [x] Kindly make sure to set up the `rentals` getters and setters, you can do this by adding the `rentals` to the `attr_accessor`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @carreraprogrammer, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Setup associations
✔️ Clean code
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (carreraprogrammer) [reply person.rb:7]: Thanks for the review @Ismailco. I'll do all the changes 😁

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require './rental'
```
Comment (rloterh) [inline book.rb:1]: - [x] [OPTIONAL]
By convention, it's a good practice to use `require_relative` classes in the same source and use `require` for external modules or packages. Kindly ensure to improve upon this aspect.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Daniel,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job implementing recommended changes!
🎲 ReadMe looks great!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 2 above)
Comment (carreraprogrammer) [reply book.rb:1]: Thank you for the sugges @rloterh, I'll follow your recomendations 😁🎊🎉

**Pull request merged**